types/block_test.go

```diff
@@ -233,6 +234,40 @@ func TestCommitValidateBasic(t *testing.T) {
 	assert.Error(t, commit.ValidateBasic())
 }
 
+func TestMaxHeaderBytes(t *testing.T) {
+	// Construct a UTF-8 string of MaxChainIDLen length using the supplementary
+	// characters.
+	// Each supplementary character takes 4 bytes.
+	// http://www.i18nguy.com/unicode/supplementary-test.html
+	maxChainID := ""
+	for i := 0; i < MaxChainIDLen; i++ {
+		maxChainID += "𠜎"
+	}
+
+	h := Header{
+		ChainID:            maxChainID,
+		Height:             10,
+		Time:               time.Now().UTC(),
+		NumTxs:             100,
+		TotalTxs:           200,
```

Comment:
Aren't these varints? What if we had `TotalTxs: 20000` - then the test fails because the header is 479 bytes